### ISSUE TITLE
operationName not set when doing fetch request

### ISSUE THREAD

**Issue by MarkMolina** [apollographql/apollo-ios#294]
After generating code based on the downloaded schema I receive query's like:

```
public static let operationDefinition =
    "query QueryName($parameters: [String]!) {" +
    "  foo(bar: $lorem) {" +
    "    __typename" +
    "    x" +
    "    y" +
    "  }" +
    "}"
```

I can then easily tell the client to do a fetch request and pass an instance of this query. It works fine but for some reason the QueryName (which is the operationName right?) is not set in the request. Using Charles Proxy I found that the iOS SDK is also not setting any operationName on the request and therefore our request are marked as "unknown" in new relic. 

If I inspect our javascript client on our website it does actually add an operationName to the request. This is done automatically. Is there a way to also do this for the iOS SDK?

Using Apollo iOS version 0.5.6.

**Comment by martijnwalraven** [comment]
Since [`operationName` is only required when multiple operations are included in a single query document](https://graphql.org/learn/serving-over-http/#post-request), and Apollo iOS never sends multiple operations, we don't currently set this. But it would be pretty easy to add to `HTTPNetworkTransport` if you have a need for it (like logging in New Relic).

**Comment by MarkMolina** [comment]
Thanks Martijn. Do you know how I can easily retrieve the query name from the generated code? I see that I can easily change the send method to include an operationName but am not sure how to retrieve the matching name that goes with that specific request.

Maybe from the `GraphQLOperation` parameter?

**Comment by martijnwalraven** [comment]
Ah, you're right, we don't currently generate an `operationName` on `GraphQLOperation` either. We could definitely do that, but it does require changes to codegen. It's a bit of a hack, but for now you might be able to derive it from `operationDefinition` (looking at `operationType` to cut off `query` or `mutation` and know where to find the name in the string).

**Comment by MarkMolina** [comment]
Right, I'll implement it like suggested for our logging purposes for now and keep an eye on any updates. Maybe I'll dive into the code as well and see I can help out with this.

**Comment by DarkDust** [comment]
According to [this document](https://graphql.org/learn/queries/): _"It is only required in multi-operation documents, but its use is encouraged because it is very helpful for debugging and server-side logging."_ 

We'd like to use them for subscriptions so support for them would be welcome. Does the note in [the draft section 5.2.3.1](https://facebook.github.io/graphql/draft/#sec-Single-root-field) mandate to send the operation name or am I misunderstanding this?

**Comment by DarkDust** [comment]
Ping @martijnwalraven : how's your take on supporting `operationName`?

**Comment by designatednerd** [comment]
This is definitely something we're looking at including -  I have to see where things are with our current setup with the typescript stuff, but this is a totally reasonable request.

**Comment by designatednerd** [comment]
This is now included with everything as of 0.13.0!
